stateTransition.go

```diff
@@ -111,3 +112,14 @@ func (s *StateTransitionPubSignals) PubSignalsUnmarshal(data []byte) error {
 func (s StateTransitionPubSignals) GetObjMap() map[string]interface{} {
 	return toMap(s)
 }
+
+func (ao *StateTransitionPubSignals) VerifyStates(ctx context.Context, stateVerFunc StateResolverFunc) error {
```

Comment:
Do we need this verification at all?

Reply:
???

Reply:
otherwise, StateTransitionPubSignals doesn't implement the interface. It can return the error 'not implemented', but I guess it's more logical to give a possibility to verify ID and State of proof if somebody for some reason needs it.

Reply:
Actually, it makes sense. It is may be better to verify state before trying to call the smart contract. It may save some gas it case if we are trying to transact with revoked private key, for example.